src/plugins/typedispatcher/helpers.c

```diff
@@ -0,0 +1,437 @@
+{
+	const char * ptr = keyString (key);
+#ifdef DEVBUILD
+	fprintf (stderr, "replacing parameters in %s\n", ptr);
+#endif
+	const char * lastPtr = ptr;
+	char * preparedString = NULL;
+	char * dstPtr = NULL;
+	ssize_t preparedStringLen = 1;
+	while (*ptr)
+	{
+		if (*ptr == '%')
+		{
+			const char * endPtr = ptr + 1;
+			if (lastPtr == ptr) ++lastPtr;
+			while (*endPtr)
+			{
+				if (*endPtr == '%')
+				{
+					// copy string from lastPtr to ptr
+					preparedStringLen += (ptr - lastPtr) + 1;
+					elektraRealloc ((void **)&preparedString, preparedStringLen);
+					if (!dstPtr)
+					{
+						dstPtr = preparedString;
+					}
+					else
+					{
+						dstPtr = preparedString + elektraStrLen (preparedString) - 1;
+					}
+					*dstPtr = '\0';
+					if (ptr >= lastPtr)
+					{
+						memcpy (dstPtr, lastPtr, (ptr - lastPtr));
+						dstPtr += (ptr - lastPtr);
+						*dstPtr = '\0';
+						lastPtr = endPtr + 1;
+					}
+
+
+					// extract argument name
+					ssize_t len = (endPtr - 1) - (ptr + 1) + 1; // length of argument string
+					char tmp[len + 1];			    // length + nullbyte
+					memset (tmp, 0, sizeof (tmp));
+					memcpy (tmp, ptr + 1, len);
+					Key * searchKey = keyNew (tmp, KEY_META_NAME, KEY_END);
+					Key * lookup = ksLookup (args, searchKey, KDB_O_NONE);
+					keyDel (searchKey);
+					if (lookup)
+					{
+
+						ssize_t newLen = preparedStringLen + elektraStrLen (keyString (lookup)) -
+								 1; // no need for 2 nullbytes
+						elektraRealloc ((void **)&preparedString, newLen);
+						strncat (preparedString, keyString (lookup), elektraStrLen (keyString (lookup)) - 1);
+						dstPtr = preparedString + newLen - 1;
+						preparedStringLen = newLen;
+					}
+					else
+					{
+						ssize_t newLen = elektraStrLen (preparedString) + sizeof (tmp) + 2;
+						elektraRealloc ((void **)&preparedString, newLen);
+						strcat (preparedString, "%");
+						strncat (preparedString, tmp, sizeof (tmp));
+						strcat (preparedString, "%");
+						dstPtr = preparedString + newLen - 1;
+						preparedStringLen = newLen;
+					}
+					ptr = endPtr;
+					lastPtr = ptr + 1;
+					break;
+				}
+				++endPtr;
+			}
+		}
+		++ptr;
+	}
+	if (!(*ptr) && lastPtr && preparedString)
+	{
+		ssize_t newLen = elektraStrLen (preparedString) + (ptr - lastPtr) + 1;
+		elektraRealloc ((void **)&preparedString, newLen);
+		strncat (preparedString, lastPtr, (ptr - lastPtr) + 1);
+	}
+
+	if (preparedString)
+	{
+#ifdef DEVBUILD
+		fprintf (stderr, "result: %s\n", preparedString);
+#endif
+		return preparedString;
+	}
+	else
+	{
+#ifdef DEVBUILD
+		fprintf (stderr, "result: %s\n", keyString (key));
+#endif
+		return elektraStrDup (keyString (key));
+	}
+}
+
+
+// parses "type (arg1, arg2, ...)" strings into ArgumentConfig
+// TODO: rewrite
+ArgumentConfig * parseTypeString (DispatchConfig * config ELEKTRA_UNUSED, const char * typeString)
+{
+	const char * ptr = typeString;
+	while (*ptr)
+	{
+		if (*ptr == '(' || *ptr == ' ') break;
```

Comment:
directly below in line 326 you have such check in your loop condition, which i also prefer.